base/set.jl

```diff
@@ -686,16 +694,34 @@ julia> replace([1, 2, 1, 3], 1=>0, 2=>4, count=2)
  3
 ```
 """
-function replace(A, old_new::Pair...; count::Integer=typemax(Int))
+function replace(A, old_new::Pair...; count::Union{Integer,Nothing}=nothing)
```

Comment:
The use of `Nothing` here for type stability is quite subtle; if you re-push to this branch for another change, may be worth a small comment to explain this.